app/services/service_catalog/submit_order.rb

```diff
@@ -35,12 +35,30 @@ def parameters(item)
       end
     end
 
-    def update_item(item, result)
-      item.topology_task_ref = result.task_id
-      item.state             = 'Ordered'
-      item.ordered_at        = Time.now.utc
-      item.update_message('info', 'Initialized')
-      item.save!
+    def update_item(item, task_id)
+      Thread.new do
+        task = check_task(task_id)
+        item.topology_task_ref = task_id
```

Comment:
@eclarizio 
Updating of the item would have to happen first so we can record that that it has been ordered.
Then the new thread monitors the task and updates the item along the way with its status.

Reply:
Something similar to this
https://github.com/ManageIQ/service_portal-api/blame/master/app/models/order_item.rb

Reply:
> Updating of the item would have to happen first so we can record that that it has been ordered.

Ok, I see, I misunderstood the "Ordered" state as being finished, not an intermediary state. Will fix.

> Then the new thread monitors the task and updates the item along the way with its status.

Ok, but the updates that comes back from the external API is the main question I have. Where does it go? `source_ref` and `source_id` that are coming back correspond to which fields on the `item`?